timer/timer.go

```diff
@@ -0,0 +1,144 @@
+package timer
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Timer definition
+type (
+	Timer struct {
+		ID             uuid.UUID
+		Domain         string
+		ExecutionCount uint32
+		Meta           Meta
+		Task           Task
+		Schedule       Schedule
+	}
+
+	// Task represents a timer task.
+	Task interface {
+		isTask()
+	}
+
+	// Schedule represents the "when" configuration of a timer.
+	Schedule interface {
+		isSchedule()
+	}
+
+	// Meta represents timer metadata.
+	Meta struct {
+		CreationTime time.Time
+	}
+)
+
+// HTTP Task represents an HTTP request.
+type (
+	HTTP struct {
+		URL string
+		Method
+		Body    string
+		Headers map[string]string
+	}
+
+	// Method is the method of the HTTP request.
+	Method int
+)
+
+func (HTTP) isTask() {}
+
+// Method definitions
+const (
+	GET Method = iota
+	POST
+)
+
+// Schedule definitions
+type (
```

Comment:
There is too much going on in this file. It contains metadata for a timer, schedule, different timer types, tasks and timers. These need to be broken apart and tested separately.

Reply:
How do I test the types, they're just structs/interfaces? I can test marshalling/unmarshalling.